dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java

```diff
@@ -688,7 +693,8 @@ public synchronized void doOverrideIfNecessary() {
                 return;
             }
             //The current, may have been merged many times
-            URL currentUrl = exporter.getInvoker().getUrl();
+            Invoker<?> exporterInvoker = exporter.getInvoker();
```

Comment:
exporterInvoker should not be null here when calling `doOverrideIfNecessary`

Reply:
one report of github say NPE here

Reply:
属于偶发异常，可能和 #8725 是同样的问题，AfterEach的释放全局资源操作干扰了正在运行的dubbo后台线程，导致后续的测试方法出现异常。

Reply:
in branch 3.0, ReferenceConfigTest testReferenceRetry()  uses InjvmProtocol.getInjvmProtocol().export(...) instead of  sc.export(), it doesn't call RegistryProtocol.doOverrideIfNecessary at all, so 3.0 might also have the same issue.